Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2733 -- Expand Search component on search page

Resolves [EREGCSC-2733](https://jiraent.cms.gov/browse/EREGCSC-2733)

**Description**

Adds a "Continue Your Research" component to the bottom of Search Results on the Search page.

**This pull request changes:**

- Renames `SearchEmptyState` component into `SearchContinueResearch` component
   - refactors code from Vue Options API to Composition API
   - Removes `style` tags and moves style rules to SCSS stylesheet
- Updates "No Results" message on Search page based on logic outlined in AC
- Adds logic to allow users to try quoted searches on multi-word queries (not on single word queries)
- Adds Vitest unit tests and Cypress e2e tests
- Removes `RegResults.vue` component that is no longer used in the app

**Steps to manually verify this change:**

1. Green check marks
2. Visit [experimental deployment](https://n9mokp6z9c.execute-api.us-east-1.amazonaws.com/dev1422)
3. Briefly compare to [Figma comp](https://www.figma.com/design/QTeJIYHaipHux8tx88A4jh/Search-Everything?m=dev) to make sure things look okay
4. Try a single word query.  Example: "test"
   - At the bottom of the list of results, the "Continue Your Research" component should be visible.  It should have a single row inviting the user to "Try your search for test on other websites."
   - These links should all take you to the expected destination with the correct search term filled in ("test").
5. Apply a filter to this same single word query ("test").  Click the "Public Resources" checkbox.
   - At the bottom of the list of results, the "Continue Your Research" component should be visible. 
   - It should now have **two** rows
   - The top row should be labeled "Broaden your search on eRegulations" and there should be a "reset all active filters" link.
   - Click the "reset all active filters" link
   - The Public Resources checkbox should no longer be checked
   - At the bottom of the list of results, the "Continue Your Research" component should be visible.  It should have a single row inviting the user to "Try your search for test on other websites."
6. Try multi-word search.  Example: "test search"
   - At the bottom of the list of results, the "Continue Your Research" component should be visible. 
   - It should now have **two** rows
   - The top row should be labeled "Make your search more specific" and there should be a link to "try your search with quotes"
   - Click the link to the right of "try your search with quotes"
   - the quoted search term should appear in the URL and in the search input
   - At the bottom of the list of results, the "Continue Your Research" component should be visible.  It should have a single row inviting the user to 'Try your search for "test search" on other websites.'
7. Try a search that is guaranteed to return no results.  Example: "asdfghjk"
   - There should be no results and you should see a message that says "Your search for asdfghjk did not match any results on eRegulations."
   - Apply a filter to this same single word query ("asdfghjk").  Click the "Public Resources" checkbox.
   - The no results message should change: it should now say "Your search for asdfghjk did not match any results with the selected filters."
